### PR TITLE
Fix for criteria update issue

### PR DESCRIPTION
**Pull request by JosteinLindhom**
- Removed Review deletion (for now).
- One will not always know the assignment ID ahead of wanting to fetch benchmarks.
- Added a test that tries to simulate UpdateFromTestsRepo, and verify that is does not change unrelated stuff
- GetReview database function
- Removed old test that is covered elsewhere

### CODE REVIEW

File: assignments/assignments_test.go [hunk 1]
```diff
@@ -34,3 +37,261 @@ func TestFetchAssignments(t *testing.T) {
...
+		t.Fatalf("Expected assignment.GradingBenchmarks to be nil, got %v", assignment.GradingBenchmarks)
+	}
+
+	// Update assignments. GradingBenchmarks should not be updated
+	db.UpdateAssignments([]*pb.Assignment{assignment, assignment2})
+
+	for _, wantReview := range []*pb.Review{review, review2} {
+		if gotReview, err := db.GetReview(&pb.Review{ID: wantReview.ID}); err != nil {
+			t.Fatal(err)
+		} else {
```
Comment (meling) [inline assignments/assignments_test.go:218]: There is no need for `else` after `t.Fatal`. Prefer [normal code path with minimal indentation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow). Applies to several cases in this test.

File: assignments/assignments_test.go [hunk 2]
```diff
@@ -34,3 +37,261 @@ func TestFetchAssignments(t *testing.T) {
...
+		t.Fatal("Expected assignment.GradingBenchmarks to not be nil")
+	}
+
+	// Update assignments. GradingBenchmarks should be updated
+	db.UpdateAssignments([]*pb.Assignment{assignment, assignment2})
+
+	// Benchmarks should have been updated to reflect the removal of a benchmark and a criterion
+	if gotBenchmarks, err := db.GetBenchmarks(&pb.Assignment{ID: assignment.ID, CourseID: course.ID}); err != nil {
+		t.Fatal(err)
+	} else {
```
Comment (meling) [inline assignments/assignments_test.go:280]: No `else`

File: assignments/assignments_test.go [hunk 3]
```diff
@@ -34,3 +37,261 @@ func TestFetchAssignments(t *testing.T) {
...
+		if diff := cmp.Diff(updatedBenchmarks, gotBenchmarks, protocmp.Transform()); diff != "" {
+			t.Errorf("GetBenchmarks() mismatch (-want +got):\n%s", diff)
+		}
+	}
+
+	// Finally check that reviews are unaffected
+	for _, wantReview := range []*pb.Review{review, review2} {
+		if gotReview, err := db.GetReview(&pb.Review{ID: wantReview.ID}); err != nil {
+			t.Fatal(err)
+		} else {
```
Comment (meling) [inline assignments/assignments_test.go:290]: No `else`

**Review by meling — CHANGES_REQUESTED** [review]
Looks good to me! Minor tweaks needed in the `_test.go` file. Also, isn't it possible to rewrite the `TestEqual` thing in `gormdb_assignment_test.go`? Fine if it doesn't make sense but just wanted to check.

**Review by meling — APPROVED** [review]

**Review by 0xf8f8ff — APPROVED** [review]
It works!

**Pull request merged**